### PR TITLE
[13.0][FIX] agreement_legal: Change create_new_version() function in agreements

### PR DESCRIPTION
**Pull request by victoralmau**
Change `create_new_version()` function in agreements + Add tests.

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT32359

### CODE REVIEW

File: agreement_legal/models/agreement.py [hunk 1]
```diff
@@ -371,7 +371,7 @@ def create_new_version(self, vals):
             # Increment the Version
             rec.version = rec.version + 1
         # Reset revision to 0 since it's a new version
-        vals["revision"] = 0
+        vals = {"revision": 0}
```
Comment (pedrobaeza) [inline agreement_legal/models/agreement.py]: I think the logic is incorrect, as you are writing `revision = 0` on the old agreement, not in the new. Is this how it's supposed to work?

Comment (victoralmau) [reply agreement_legal/models/agreement.py]: Yes, the logic seems wrong. You should define `revision = 0` for new records (child records in these case).
Thinking about this issue, IMO increase the revision is not correct when performing this action. What do you think?

Comment (pedrobaeza) [reply agreement_legal/models/agreement.py]: How does this work in v12?

Comment (victoralmau) [reply agreement_legal/models/agreement.py]: In v12 it has the strange behavior.
https://github.com/OCA/contract/blob/9893ba7589c67451e7f2cc071b2ca211d2706f9b/agreement_legal/models/agreement.py#L319-L336

Comment (victoralmau) [reply agreement_legal/models/agreement.py]: Changes done to standardize this behavior and show "archived" versions.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Fast-tracking for fixing the bug:

/ocabot merge patch

**Pull request merged**